tests/test_coreg/test_affine.py

```diff
@@ -596,3 +596,28 @@ def test_nuthkaab_no_vertical_shift(self) -> None:
         # Assert horizontal shifts are the same
         matrix2[2, 3] = matrix1[2, 3]
         assert np.array_equal(matrix1, matrix2)
+
+    def test_nuthkaab_initial_shift(self) -> None:
```

Comment:
Same comment here, we can test this works well for every affine method, so just Nuth and Kääb!

Reply:
For the tests, it's only adding `@pytest.mark.parametrize("coreg_method", [all_coreg_methods])` everywhere, and for the code itself within `fit()`, it's what I wrote above.
Probably less than 10 lines of code overall, so I think it's shorter to do it here? 

Happy to provide more help on a new `set_matrix()` as I described above, if you want. Or take over the PR.